### PR TITLE
Modernizing xcm config for shellruntime

### PR DESCRIPTION
**Pull request by coax1d**
- Having Shell runtime have configured xcm.

### CODE REVIEW

File: Cargo.lock [hunk 1]
```diff
@@ -9885,13 +9886,17 @@ dependencies = [
  "hex",
  "hex-literal",
  "log",
+ "orml-traits",
+ "orml-xcm",
+ "orml-xcm-support",
  "pallet-aura",
```
Comment (clangenb) [inline Cargo.lock:9892]: Nice, this lock file confirms that we do not pull in the duplicate substrate deps from the dev-dependencies.

File: polkadot-parachains/shell-runtime/src/lib.rs [hunk 2]
```diff
@@ -28,11 +28,18 @@ use sp_api::impl_runtime_apis;
...
 	ApplyExtrinsicResult,
 };
 
+// use sp_runtime::{
+// 	create_runtime_str, generic, impl_opaque_keys,
+// 	traits::{AccountIdLookup, BlakeTwo256, Block as BlockT, Convert, ConvertInto},
+// 	transaction_validity::{TransactionSource, TransactionValidity},
+// 	ApplyExtrinsicResult,
+// };
+
```
Comment (clangenb) [inline polkadot-parachains/shell-runtime/src/lib.rs:42]: If this is not needed, I would like to ask you to remove this.

**Review by clangenb — APPROVED** [review]
Cool, this looks great. I have not checked all the details regarding the XCM config, but as you said that you mostly copied it from the integritee-runtime I suppose this is fine.

I have only one minor remark regarding some commented code. Hence, I will already approve, and you can merge it after you fixed the comment.

File: Cargo.lock (hunk 1 above)
Comment (clangenb) [reply Cargo.lock:9892]: Ah wait, this is not relevant yet, we have to hope that we don't see the deps in the other PR. :)

File: polkadot-parachains/shell-runtime/src/lib.rs (hunk 2 above)
Comment (coax1d) [reply polkadot-parachains/shell-runtime/src/lib.rs:42]: Yea snuck in somehow

File: polkadot-parachains/shell-runtime/src/lib.rs [hunk 3]
```diff
@@ -342,7 +288,10 @@ construct_runtime!(
 		Aura: pallet_aura::{Pallet, Storage, Config<T>} = 23,
 		AuraExt: cumulus_pallet_aura_ext::{Pallet, Config} = 24,
 
-		CumulusXcm: cumulus_pallet_xcm::{Pallet, Call, Storage, Event<T>, Origin} = 32,
+		XcmpQueue: cumulus_pallet_xcmp_queue::{Pallet, Call, Storage, Event<T>} = 30,
+		PolkadotXcm: pallet_xcm::{Pallet, Call, Event<T>, Origin} = 31,
+		CumulusXcm: cumulus_pallet_xcm::{Pallet, Call, Event<T>, Origin} = 32,
```
Comment (clangenb) [inline polkadot-parachains/shell-runtime/src/lib.rs:293]: Great, you check the details. :)

**Review by clangenb — APPROVED** [review]
Thanks for the fix, looks good to me now! 🚀

**Review by clangenb — APPROVED** [review]

**Pull request merged**